components/camera/camera.go

```diff
@@ -138,20 +126,44 @@ type ImagesSource interface {
 // If needed, implement the Camera another way. For example, a webcam
 // implements a Camera manually so that it can atomically reconfigure itself.
 func FromVideoSource(name resource.Name, src VideoSource, logger logging.Logger) Camera {
+	var rtpPassthroughSource rtppassthrough.Source
+	if ps, ok := src.(rtppassthrough.Source); ok {
+		rtpPassthroughSource = ps
+	}
 	return &sourceBasedCamera{
-		Named:       name.AsNamed(),
-		VideoSource: src,
-		Logger:      logger,
+		rtpPassthroughSource: rtpPassthroughSource,
+		Named:                name.AsNamed(),
+		VideoSource:          src,
+		Logger:               logger,
 	}
 }
 
 type sourceBasedCamera struct {
 	resource.Named
 	resource.AlwaysRebuild
 	VideoSource
+	rtpPassthroughSource rtppassthrough.Source
 	logging.Logger
 }
 
+func (vs *sourceBasedCamera) SubscribeRTP(
```

Comment:
This is needed for cameras that  are wrapped in `FromVideoSource` to still implement `rtppassthrough.SubscriptionID`